### PR TITLE
S2 | Enchanced signup route #28

### PR DESCRIPTION
**Pull request by shalayiding**
done

### CODE REVIEW

**Comment by shalayiding** [comment]
do i have to change anything ? @raghunat

File: users-api/routes/signup.js [hunk 1]
```diff
@@ -14,17 +14,48 @@ router.get('/testRoute', (req,res) => res.send('Test successful.'));
...
 // confirm the signup and display the information
 router.post('/', (req, res) => {
-  if((!req.body.username) || (!req.body.password))
+  if(req.body.username)
   {
-    res.status(403).send('Invalid information.');
-  }
+  if((!req.body.username) || (!req.body.password))
+    {
+   res.status(403).send('Invalid information.');
```
Comment (raghunat) [inline users-api/routes/signup.js]: We need to use `res.status(403).json({message: "Invalid Information"})` for our UI to parse it correctly

File: users-api/routes/signup.js [hunk 2]
```diff
@@ -14,17 +14,48 @@ router.get('/testRoute', (req,res) => res.send('Test successful.'));
...
+    var password=req.body.password;
+    var Email_Check=/^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+    var Email=req.body.email;
+
+      req.body.id="abcd-efgh-ijkl-mnop";//-->testing id
+
+
+    if((!password)||password.length < 6||!password.match(/[a-z][0-9]+/i)||!Email_Check.test(Email))
+    {
+      res.status(401).send('"message": "Invalid Credentials"');
```
Comment (raghunat) [inline users-api/routes/signup.js]: Same thing here as above

File: users-api/routes/signup.js [hunk 3]
```diff
@@ -14,17 +14,48 @@ router.get('/testRoute', (req,res) => res.send('Test successful.'));
...
+    }
+    else{
+      res.status(200).send(
```
Comment (raghunat) [inline users-api/routes/signup.js]: Samething here

**Comment by shalayiding** [comment]
if((!password)||password.length < 6||!password.match(/[a-z][0-9]+/i)||!Email_Check.test(Email))
    {
      res.status(401).json({message: "Invalid Information"});
    }
    else{
      res.status(200).json({id: "abcd-efgh-ijkl-mnop","email":req.body.email});


    }

**Comment by shalayiding** [comment]
is this correct ?

**Comment by raghunat** [comment]
@shalayiding Close:

This is more inline with what is intended:

```js
router.post('/', (req, res) => {
  if (
    !req.body.password ||
    !req.body.email ||
    req.body.password.length < 6 ||
    req.body.email.match(/^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/) == false ||
    req.body.password.match(/[a-z0-9]+/i) == false
  ) {
    res.status(401).json({message: "Invalid Credentials"})
  } else {
    res.json({id: "abcd-efgh-ijkl-mnop", "email": req.body.email});
  }
});
```

**Comment by raghunat** [comment]
@shalayiding Up on this!

**Comment by raghunat** [comment]
Comments still not resolved

**Pull request closed without merging**